src/LanguageServer/Impl/Indexing/IndexManager.cs

```diff
@@ -57,18 +57,18 @@ internal class IndexManager : IIndexManager {
             _disposables
                 .Add(_symbolIndex)
                 .Add(() => idleTimeService.Idle -= OnIdle);
-
-            StartAddRootDir();
         }
 
         public int ReIndexingDelay { get; set; } = DefaultReIndexDelay;
 
-        private void StartAddRootDir() {
-            foreach (var fileInfo in WorkspaceFiles()) {
-                if (ModulePath.IsPythonSourceFile(fileInfo.FullName)) {
-                    _symbolIndex.Parse(fileInfo.FullName);
+        public Task IndexWorkspace(CancellationToken ct = default) {
+            return Task.Run(() => {
+                foreach (var fileInfo in WorkspaceFiles()) {
+                    if (ModulePath.IsPythonSourceFile(fileInfo.FullName)) {
```

Comment:
Doesn't seem to be using the `ct`. Should be doing then `ct.ThrowIfCancellationRequested` in the loop.